frontends/p4-14/p4-14-parse.ypp

```diff
@@ -824,31 +824,31 @@ expression:
     | LATEST '.' name   { $$ = new IR::Member(@1+@3, new IR::NamedRef(@1, $1), IR::ID(@3, $3)); }
     | INTEGER           { $$ = $1; }
     | '(' expression ')'                { $$ = $2; }
-    | NOT expression %prec PREFIX       { $$ = new IR::LNot(@1, $2); }
-    | '!' expression %prec PREFIX       { $$ = new IR::LNot(@1, $2); }
-    | '~' expression %prec PREFIX       { $$ = new IR::Cmpl(@1, $2); }
-    | '-' expression %prec PREFIX       { $$ = new IR::Neg(@1, $2); }
+    | NOT expression %prec PREFIX       { $$ = !$2; }
```

Comment:
This is an odd difference between the P4_14 and _16 frontends that I hadn't noticed before -- the P4_14 frontend always set the source info to the full expression (not just the operator token).  I'm not sure which is better aesthetically for error messages.  We could change it back to explicity setting the source info to just the operator char if desired.